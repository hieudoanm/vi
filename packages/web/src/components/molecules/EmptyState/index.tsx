import Button from '@mui/material/Button';
import React from 'react';

export type EmptyStateProps = {
  icon?: string | JSX.Element;
  title?: string;
  description?: string;
  button?: {
    name: string;
    onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
    text: string | JSX.Element;
  };
};

export const EmptyState: React.FC<EmptyStateProps> = ({
  icon = '',
  title = '',
  description = '',
  button = { name: '', text: '', onClick: () => {} },
}) => {
  return (
    <div className="rounded-lg border border-dashed	p-8">
      {icon && <div className="text-center pb-4">{icon}</div>}
      <div className="text-center pb-8">
        <h3 className="font-semibold text-2xl uppercase">{title}</h3>
        <p className="text-gray-500">{description}</p>
      </div>
      <Button type="button" name={button.name} onClick={button.onClick}>
        {button.text}
      </Button>
    </div>
  );
};

EmptyState.displayName = 'EmptyState';
EmptyState.defaultProps = {};

export default EmptyState;
