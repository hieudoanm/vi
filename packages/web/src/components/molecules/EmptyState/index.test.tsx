import EmptyState from '.';
import { render } from '@testing-library/react';

describe('EmptyState', () => {
  test('render default', () => {
    const wrapper = render(
      <EmptyState title="title" description="description" />
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
