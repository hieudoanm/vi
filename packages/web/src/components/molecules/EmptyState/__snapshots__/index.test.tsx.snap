// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState render default 1`] = `
<div>
  <div
    class="rounded-lg border border-dashed p-8"
  >
    <div
      class="text-center pb-8"
    >
      <h3
        class="font-semibold text-2xl uppercase"
      >
        title
      </h3>
      <p
        class="text-gray-500"
      >
        description
      </p>
    </div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      name=""
      tabindex="0"
      type="button"
    >
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
